apps/api/src/app/events/usecases/send-message/digest/get-digest-events.usecase.ts

```diff
@@ -6,19 +6,19 @@ import {
   DetailEnum,
 } from '../../../../execution-details/usecases/create-execution-details/create-execution-details.command';
 import { CreateExecutionDetails } from '../../../../execution-details/usecases/create-execution-details/create-execution-details.usecase';
+import { DigestFilterSteps } from '../../digest-filter-steps/digest-filter-steps.usecase';
 
 @Injectable()
 export abstract class GetDigestEvents {
   constructor(protected jobRepository: JobRepository, private createExecutionDetails: CreateExecutionDetails) {}
 
   protected async filterJobs(currentJob: JobEntity, transactionId: string, jobs: JobEntity[]) {
-    const batchValue = currentJob?.payload ? currentJob.payload[currentJob?.digest?.digestKey] : undefined;
-    if (batchValue) {
-      jobs = jobs.filter((job) => {
-        return job.payload[currentJob.digest.digestKey] === batchValue;
-      });
-    }
-
+    const batchValue = currentJob?.payload
+      ? DigestFilterSteps.getNestedValue(currentJob.payload, currentJob?.digest?.digestKey)
+      : undefined;
+    jobs = jobs.filter((job) => {
```

Comment:
You removed by accident the check of the condition if `batchValue` exists that avoids to run the loop as it would be useless.

Reply:
No @p-fernandez This is removed intentionally, batchvalue missing jobs should have separate digest